### PR TITLE
Initial guide >= stub. Includes hints.

### PR DESCRIPTION
**Pull request by jamesliudotcc**
This is a pretty easy problem, so I had problems coming up with hints that did not give the problem away.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #XXXXX

### CODE REVIEW

**Comment by RandellDawson** [comment]
Closed and Reopened this PR to attempt to resolve a specific Travis build failure.

File: guide/english/certifications/apis-and-microservices/apis-and-microservices-projects/request-header-parser-microservice/index.md [hunk 1]
```diff
@@ -2,9 +2,67 @@
...
+## ![:clipboard:](https://forum.freecodecamp.com/images/emoji/emoji_one/clipboard.png?v=3 ':clipboard:') NOTE TO CONTRIBUTORS:
 
-<a href='https://github.com/freecodecamp/guides/blob/master/README.md' target='_blank' rel='nofollow'>This quick style guide will help ensure your pull request gets accepted</a>.
+- ![:warning:](https://forum.freecodecamp.com/images/emoji/emoji_one/warning.png?v=3 ':warning:') **DO NOT** add solutions that are similar to any existing solutions. If you think it is **_similar but better_**, then try to merge (or replace) the existing similar solution.
+- Add an explanation of your solution.
+- Categorize the solution in one of the following categories -- **Basic**, **Intermediate** and **Advanced**. ![:traffic_light:](https://forum.freecodecamp.com/images/emoji/emoji_one/traffic_light.png?v=3 ':traffic_light:')
+- Please add your username only if you have added any **relevant main contents**. (![:warning:](https://forum.freecodecamp.com/images/emoji/emoji_one/warning.png?v=3 ':warning:') **_DO NOT_** _remove any existing usernames_)
```
Comment (thecodingaviator) [inline guide/english/certifications/apis-and-microservices/apis-and-microservices-projects/request-header-parser-microservice/index.md]:
```suggestion
```

File: guide/english/certifications/apis-and-microservices/apis-and-microservices-projects/request-header-parser-microservice/index.md [hunk 2]
```diff
@@ -2,9 +2,67 @@
...
 
-<!-- The article goes here, in GitHub-flavored Markdown. Feel free to add YouTube videos, images, and CodePen/JSBin embeds  -->
+> See ![:point_right:](https://forum.freecodecamp.com/images/emoji/emoji_one/point_right.png?v=3 ':point_right:') <a href='http://forum.freecodecamp.com/t/algorithm-article-template/14272' target='_blank' rel='nofollow'>**`Wiki Challenge Solution Template`**</a> for reference.
```
Comment (thecodingaviator) [inline guide/english/certifications/apis-and-microservices/apis-and-microservices-projects/request-header-parser-microservice/index.md]:
```suggestion
```

**Review by thecodingaviator — DISMISSED** [review]

File: guide/english/certifications/apis-and-microservices/apis-and-microservices-projects/request-header-parser-microservice/index.md [hunk 3]
```diff
@@ -2,9 +2,65 @@
...
+
+```javascript
+app.get('/api/whoami/', (req, res) => {
+  let responseObject = {
+    ipaddress: req.ip,
+    language: req.get('accept-language'),
+    software: req.get('user-agent'),
+  };
+
+  res.json(responseObject);
```
Comment (thecodingaviator) [inline guide/english/certifications/apis-and-microservices/apis-and-microservices-projects/request-header-parser-microservice/index.md]:
```suggestion
res.json(responseObject);
```

**Comment by RandellDawson** [comment]
@thecodingaviator Are you ok with the committed changes?  If so, please merge.  Thanks.

**Comment by thecodingaviator** [comment]
@RandellDawson I haven’t tested the solution yet. Have you by any chance done it?

**Comment by RandellDawson** [comment]
Thank you for opening this pull request.

This is a standard message notifying you that we’ve reviewed your pull request and have decided not to merge it. We would welcome future pull requests from you.

Reason:  This is not a full working solution.  It returns the ip address "127.0.0.1" instead of the user's ip address.

Thank you and happy coding.

**Pull request closed without merging**